src/reportengine/api.py

```diff
@@ -0,0 +1,47 @@
+"""
+API for resource engine
+"""
+
+import logging
+import importlib
+
+from reportengine.resourcebuilder import ResourceBuilder
+from reportengine.resourcebuilder import FuzzyTarget
+from reportengine.configparser import Config
+from reportengine.environment import Environment
+
+log = logging.getLogger(__name__)
+
+class API:
+    """The API class"""
+    config_class = Config
+    environment_class = Environment
+
+    def __init__(self, providers, **kwargs):
+        #TODO: need to add providers here
+        prov_list = []
+        for prov in providers:
+            try:
+                mod = importlib.import_module(prov)
+            except ImportError:
+                #TODO: the code this is copying is wrong and should be changed to have prov here
+                log.error("Could not import module %s", prov)
```

Comment:
I don't know if you want it done in this PR but I think in the part of report engine which loads modules this is:
`log.error("Could not import module %s", mod)` which then incorrectly lists the previous module as not loading (since `mod = importlib.import_module(prov)` failed)

Reply:
Yeah, please fix it.